docs/kyma/docs/037-gs-installation-overrides.md

```diff
@@ -0,0 +1,168 @@
+
+
+#### Sub-chart overrides
+
+Overrides for sub charts follow the same convention as top-level charts, but require additional information about sub-chart location.
+When a sub-chart contains `values.yaml` file, the information about chart location is not necessary because chart and it's `values.yaml` file are on the same level in directory hierarchy.
+The situation is different when the Installer installs a chart with sub-charts.
+All template values for a sub-chart must be prefixed with a sub-chart "path" (relative to top-level "parent" chart).
+This in not Installer-specific requirement, the same considerations apply when you provide overrides manually via `helm` command-line tool.
+
+Example:
+
+There's a `core` top-level chart, that is installed by the Installer.
+There's an `application-connector` sub-chart in `core` with another nested sub-chart: `connector-service`.
+In one of it's templates there's a following fragment (shortened):
+```
+spec:
+  containers:
+  - name: {{ .Chart.Name }}
+	args:
+	  - "/connectorservice"
+	  - '--appName={{ .Chart.Name }}'
+	  - "--domainName={{ .Values.global.domainName }}"
+	  - "--tokenExpirationMinutes={{ .Values.deployment.args.tokenExpirationMinutes }}"
+```
+
+The following fragment of `values.yaml` file in `connector-service` chart defines default value for `tokenExpirationMinutes`:
+```
+deployment:
+  args:
+    tokenExpirationMinutes: 60
+```
+
+If you want to override this value, for example to "90", do as follows:
+- Create a ConfigMap in kyma-installer namespace, labelled with: `installer:overrides` (or reuse existing one)
+- Add an entry `application-connector.connector-service.deployment.args.tokenExpirationMinutes: 90` to the map.
+
+Notice that the user-provided override key is now composed from two parts:
+  - Chart "path" inside top-level `core` chart: **application-connector.connector-service**
+  - Original template value reference from the chart (without .Values. prefix): **deployment.args.tokenExpirationMinutes**
+Once the installation starts, Installer generates overrides based on the map entries and value of "90" will be used instead of default "60" from the chart `values.yaml` file.
+
+
+#### Global overrides
+
+Some important parameters are usually shared across the charts.
+Helm convention to provide these is by using `global` override key.
+
+For example, to define `global.domain` override, just use "global.domin" as the name of the key in ConfigMap/Secret for Installer.
+Once the installation starts, Installer is merging all map entries and collect all global entries together (under `global` top-level key) for the purpose of installation.
+
+
+## Merging and conflicting entries
+
+When installer encounters two overrides with the same key prefix, it tries to merge them.
```

Comment:
When the Installer encounters two overrides with the same key prefix, it tries to merge them.
If both of them represent a map (they have nested sub-keys), their nested keys are recursively merged.
If at least one of keys points to a final value, the Installer performs the merge in a non-deterministic order, so either one of the overrides is rendered in the final yaml file.

It is important to avoid overrides having the same keys for final values.